src/components/Upgrade/UpgradeModal.js

```diff
@@ -10,134 +10,138 @@ import ProgressBar from './ProgressBar'
-                      />
-                    )
-                  /* eslint-enable react/prop-types */
-                  }
-                </Transition>
-              </div>
-
-              <div
-                css={`
-                  position: relative;
-                  z-index: 2;
-                `}
+                css="position: relative"
                 style={{
-                  width: verticalMode ? '100%' : `${RATIO_LEFT * 100}%`,
+                  height: verticalMode
+                    ? `${height - 40}px`
+                    : `${Math.max(500, Math.min(620, height - 40))}px`,
                 }}
               >
-                <Navigation
-                  step={step}
-                  steps={steps}
-                  onPrev={prev}
-                  onNext={next}
-                />
+                <ProgressBar value={(step + 1) / steps} />
+
+                <ButtonIcon
+                  label="Close"
+                  onClick={onClose}
+                  css={`
+                    position: absolute;
+                    top: 17px;
+                    right: 17px;
+                    z-index: 2;
+                  `}
+                >
+                  <img src={closeSvg} alt="" />
+                </ButtonIcon>
+
+                <div
+                  css={`
+                    position: relative;
+                    z-index: 1;
+                    width: 100%;
+                    height: 100%;
+                    overflow: hidden;
+                    display: flex;
+                  `}
+                >
+                  <Transition
+                    native
+                    items={step}
+                    from={{
+                      enterProgress: 1 * direction,
+                      showProgress: 0,
+                    }}
+                    initial={{
+                      enterProgress: 0,
+                      showProgress: 1,
+                    }}
+                    enter={{
+                      enterProgress: 0,
+                      showProgress: 1,
+                    }}
+                    leave={{
+                      enterProgress: -1 * direction,
+                      showProgress: 0,
+                    }}
+                    config={springs.smooth}
+                  >
+                    {(index, state) =>
+                      /* eslint-disable react/prop-types */
+                      ({ enterProgress, showProgress, status }) => (
+                        <HighlightScreen
+                          compactMode={compactMode}
+                          onUpgrade={onUpgrade}
+                          verticalMode={verticalMode}
+                          enterProgress={enterProgress}
+                          showProgress={showProgress}
+                          state={state}
+                          {...highlights[index]}
+                          {...(canUpgradeOrg ? highlights['0.6'][index] : {})}
```

Comment:
We should do this merging when defining the highlights instead

Reply:
The way this is implemented allows to use other content and just have some overrides, if I follow your approach I would have to duplicate content.